projects/observability/src/shared/components/cartesian/chart-sync-service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { Band, CartesianSelectedData, Series } from '../../../public-api';
+import { MouseLocationData } from '../utils/mouse-tracking/mouse-tracking';
+
+@Injectable({ providedIn: 'root' })
+export class ChartSyncService<TData> {
+  private readonly locationChangeSubject: Subject<ChartHoverData<TData>> = new Subject();
+
+  public mouseLocationChange(
+    data: MouseLocationData<TData, Series<TData> | Band<TData>>[] | CartesianSelectedData<TData>,
+    groupId?: string
+  ): void {
+    this.locationChangeSubject.next({
+      groupId: groupId,
+      locationData: data as MouseLocationData<TData, Series<TData> | Band<TData>>[]
+    });
+  }
+
+  public getLocationChangesForGroup(groupId: string): Observable<ChartHoverData<TData>> {
+    return this.locationChangeSubject.pipe(filter(data => data.groupId === groupId));
```

Comment:
group ID doesn't need to come back on result, can pick the location data back off the object via map.